develop/application-devs/debugging-errors.md

```diff
@@ -1,72 +1,105 @@
+
+### View Error Details
+
+The runtime errors for a particular pallet can be viewed in one of several ways:
 
 1. Referring to the [`polkadot_sdk_docs`](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/index.html){target=\_blank} for that pallet, where the errors can be found, for example, [`pallet_balances`](https://paritytech.github.io/polkadot-sdk/master/pallet_balances/pallet/enum.Error.html#variants){target=\_blank}
 2. Referring to [Subscan's runtime section](https://polkadot.subscan.io/runtime){target=\_blank}, where the pallet can be selected and viewed
+
+### Deciphering Dispatch Errors
+
+If you receive a dispatch error, it will include the index of the pallet from which the error originated and the specific index of the error within that pallet's error definitions. Both of these indices start counting from zero.
+
+Runtime errors can be located by examining the runtime's codebase. For Polkadot, this involves checking the [Polkadot Fellows Runtime repository](https://github.com/polkadot-fellows/runtimes/){target=\_blank} and the [Polkadot SDK](https://github.com/paritytech/polkadot-sdk){target=\_blank} repository.
+
+As an example, consider the following dispatch error:
+
+```js
+{
+  Module: {
+    index: 7
+    error: 0x07000000
+  }
+}
+```
+
+You can see that the module index number is `7`, but the error is a hex string that you must convert to a decimal. The first byte, `07`, represents the index of the error. You'll need to convert `0x07` to a decimal, which ends up being `7`. So now you know that `7` is the index of the pallet, and `7` is the index of the error for that pallet in this example.
+
+You can view a list of the runtime's pallets and associated indices in the Polkadot Fellows Runtimes repository in the [`relay/polkadot/src/lib.rs` file](https://github.com/polkadot-fellows/runtimes/blob/main/relay/polkadot/src/lib.rs){target=\_blank}. You'll look in this file for `pub enum Runtime`, which is where you'll see each pallet and its associated index. You'll find that the Staking pallet is at index `7`.
+
+Now that you know the error occurred in the Staking pallet, you can look at the Staking pallet's code, located in the [`substrate/frame/staking/` directory](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/staking/){target=\_blank} of the Polkadot SDK repository. The errors are defined in the [`substrate/fram/staking/src/pallet/mod.rs` file](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/staking/src/pallet/mod.rs){target=\_blank} for the Staking pallet specifically. If you look for `pub enum Error` and look at the eighth error, which corresponds to the seventh index, you'll see `InsufficientBond`. So, now you know that the error you're seeing is an insufficient bond error being emitted from the Staking pallet.
```

Comment:
@eshaben While it's probably fine to reference the codebase directly, you can also opt for a more friendly option of the rust docs, i.e.,: https://docs.rs/pallet-staking/latest/pallet_staking/enum.Error.html#variants

Reply:
I'm not super familiar with the Rust docs, does the convention always follow the pallet_* (insert format name) format? I want to ensure users can easily adapt this guide for errors occurring in different pallets, so I think defining a straightforward way for them to locate the relevant pallet documentation would be helpful